ext/standaloneusers/sql/auto_install.sql

```diff
@@ -1,113 +0,0 @@
--- +--------------------------------------------------------------------+
--- | Copyright CiviCRM LLC. All rights reserved.                        |
--- |                                                                    |
--- | This work is published under the GNU AGPLv3 license with some      |
--- | permitted exceptions and without any warranty. For full license    |
--- | and copyright information, see https://civicrm.org/licensing       |
--- +--------------------------------------------------------------------+
---
--- Generated from schema.tpl
--- DO NOT EDIT.  Generated by CRM_Core_CodeGen
---
--- /*******************************************************
--- *
--- * Clean up the existing tables - this section generated from drop.tpl
--- *
--- *******************************************************/
-
-SET FOREIGN_KEY_CHECKS=0;
-
-DROP TABLE IF EXISTS `civicrm_user_role`;
-DROP TABLE IF EXISTS `civicrm_uf_match`;
```

Comment:
@ufundo I think this is the missing bit. We could just add this line into the extension's installer somewheres.

Reply:
oh wow ok... so it is currently just nuking any existing uf_match table...


to my mind that seems a bit dangerous? i know currently standaloneusers is the only game in town on standalone, so in practice not a problem, but say you're migrating to standalone from another UF, it would be pref not to nuke that table maybe?

Reply:
just wondering more generally, has the equivalent behaviour to this line been dropped if i install an extension that provides an entity in the new way? was that deliberate?

Reply:
Sorry I know its tangential, but was thinking 'if we don't do this table now, it will be 5 years down the line and someone will be cursing the "clever" reuse of the same table in a different way'.

So I gave it a go: https://github.com/civicrm/civicrm-core/commit/ea35eaf7dc67ef265661ca98d9a74bd9ccc07b44
Thanks to Api4 / this new schema it wasn't so bad... and it seems to install cleanly :)

It slightly replaces one set of exceptions with another - but I do think it makes more sense that Standalone treats the `civicrm_uf_match` table the same way as the other UFs. I feel like we're setting ourselves up for strange bugs otherwise.  E.g. if the entity spec for existing columns in `UFMatch.entityType.php` and `Users.entityType.php` ever diverged....

One thing I couldn't iron out though: I can't `User::get(FALSE)->addSelect(`contact_id.display_name`)` now that contact_id is a "magic" field. It's not terminal, but you lose this column on http://standalone.localhost:7979/civicrm/admin/users. I think it's a general Api4 thing though - it's the same with `Activity::get(FALSE)->addSelect('source_contact_id.display_name')`. Any idea if that's fixable?

Reply:
> "yes, table dropping has been dropped"

Makes a lot of sense to me! The loaded gun analogy :laughing: 

I remember worrying in the past that if you had a table name clash between two extensions, installing one would destroy all the data from the other. 

( Could we one day prefix extension entity table names with their extension name :thought_balloon: )

Reply:
I guess given the substance of table change it probably *should* be a separate PR.

Tried just dropping the core table in the extension installer to get this passing: https://github.com/ufundo/civicrm-core/commit/4b6cd4e83e63a7b54f7d85ec79ee93bb03d303f8

But it doesn't seem to work for me. Trying to call timezone function from the extension before its available by the looks of it... Investigating..

Reply:
Ah, looking at `AutomaticUpgrader` it looks like there's no intended way for the extension to hook in _before_ the table schema creation... so.. not sure what the "easy" answer is

Reply:
Ah, looking at `AutomaticUpgrader` it looks like there's no intended way for the extension to hook in _before_ the table schema creation... so.. not sure what the "easy" fix for this PR is

Reply:
@ufundo I was looking at that too. It looks like this extension was hijacking an internal callback (`onInstall`) which is not really intended to be implemented by extensions, but was the only expedient way to hook in *before* tables are created... but @totten's new automaticUpgrader changes the way these  callbacks work so that now it fires *after* the table is created.

I think I've solved it by adding a new *public* callback for `preInstall`, so that hijacking is no longer necessary.